app/assets/javascripts/models/beta/story.js

```diff
@@ -141,24 +146,60 @@ export const toggleStory = (story) => {
   };
 };
 
-const stateFor = (story) => {
-  if (story.state === "unscheduled" && typeof story.estimate === "number") return "unstarted";
-  if (story.state === "unstarted" && isUnestimatedFeature(story)) return "unscheduled";
-  if (!isFeature(story) && story.state === "unscheduled") return "unstarted";
-  return story.state;
+const isUnstartedState = (story, newAttributes) => 
+  !haveEstimate(story) && haveEstimate(newAttributes)
+
+const isUnscheduledState = (story, newAttributes) => 
+  (
+    isFeature(story._editing) && (
+      (
+        !haveEstimate(story._editing) &&
+        !haveEstimate(newAttributes)
+      ) || (
+        !haveEstimate(newAttributes) &&
+        has(newAttributes, 'estimate')
+      ) || (
+        isUnscheduled(story._editing) && 
+        !has(newAttributes, 'state') &&
+        !haveEstimate(newAttributes)
+      ) || isUnscheduled(newAttributes)
+    )
+  )
+
+const stateFor = (story, newAttributes, newStory) => {
+  const { UNSTARTED, UNSCHEDULED } = status;
+
+  if (isUnstartedState(story._editing, newAttributes)) return UNSTARTED;
+  if (isUnscheduledState(story, newAttributes)) return UNSCHEDULED;
+  
+  return newStory.state;
 }
 
+const estimateFor = (story, newAttributes, newStory) => {
+  if (isNoEstimated(story, newAttributes) || isUnscheduledState(story, newAttributes)) return '';
+  if (isFeature(newAttributes) && !isUnscheduled(story._editing)) return 1;
+  
+  return newStory.estimate;
+}
+
+const isEstimable = story => isFeature(story)
+
+const isNoEstimated = (story, newAttributes) => 
+  (!isEstimable(story._editing) && !has(newAttributes, 'storyType')) || 
+  (!isEstimable(newAttributes) && has(newAttributes, 'storyType'))
+
+const haveEstimate = story => story.estimate
```

Comment:
You could rewrite this function like this:


```js
import { property } from 'underscore';

const haveEstimate = property('estimate');
```

Reply:
I believe the right would be `hasEstimate`.

Another thing, what this function does exactly? check if has the property or get the value? 
In both cases I would rewrite:
Check if has property:
```js
import { has } from 'underscore';

const haveEstimate = story => has(story, 'estimate');
```
Get propperty:
```js
import { property } from 'underscore';

const getEstimate = property('estimate');
```

Reply:
Well, the right name for function is hasEstimate how you said because we need to check if has the property, but we have a problem if we use:
 `const hasEstimate = story => has(story, 'estimate');`
Some cases the property exists, but the value is `null` or `''`, so I thought in some solutions, but I don't know what is the best practice:

`const hasEstimate = story => Boolean(story.estimate);`
`const hasEstimate = story => !!story.estimate;`
```
import { isNull } from 'underscore';
const hasEstimate = story => has(story, 'estimate') && !isNull(story.estimate);
```

Reply:
Okay, in this case, I would use:
```js
const hasEstimate = story => Boolean(story.estimate);
```
I believe is simpler than using `undescore` in this case.
:D

Reply:
Ok, thanks!